packages/live-share-canvas/src/core/LiveCanvas.ts

```diff
@@ -55,18 +56,29 @@ enum InkingEventNames {
 /**
  * Encapsulates information about a user.
  */
-export interface IUserInfo {
+export type IUserInfo = IValidatedUserInfo & IEventUserInfo;
+
+/**
+ * Encapsulates non spoofable information about a user.
+ */
+export interface IValidatedUserInfo {
     /**
      * Optional. The user's display name.
      */
     displayName?: string;
+}
+
+/**
+ * Encapsulates information about a user that is okay to be sent through events.
+ */
+export interface IEventUserInfo {
```

Comment:
I am not sure I understand the split of `IUserInfo` into two interfaces. Both `displayName` and `pictureUrl` are optional, IUserInfo is only used to instantiate a `LiveCursor` class, and `LiveCanvas` owns building the IUserInfo object, so it will always pass it the correct `displayName`. Unless I'm misreading your changes, the split isn't necessary and you can (should) simply keep the existing `IUserInfo` interface.

Reply:
In James's changes, we are not broadcasting the `displayName` in the socket event anymore, since that is untrustworthy & easily spoofed. `LiveShareClient.getClientInfo()` checks all clientId (local or non-local), so `getLocalUserInfo` isn't really the same thing. He separated them because `LiveShareClient.getClientInfo()` does not yet support profile pictures, and thus that still gets broadcasted through the socket (though there is an argument for removing that until it does, since it is easily spoofable).

Reply:
`onGetLocalUserInfo` is supposed to allow the developer to provide their own display name and picture url. With these changes, you will always ignore the `displayName` provided by the developer and use your own instead. Thus, that interface split and change in the definition of `IUserInfo` only serves as a way to seemingly not break the consuming app while you are actually breaking it if the developer is using `onGetLocalUserInfo` and is returning a custom `displayName`... in short, you're hiding a breaking change.

The right thing to do here is to introduce an explicit breaking change and force the developer to update their code.  There is no need to change `IUserInfo` because it already accdurately represents the data that needs to be transmitted via the event. You just need to change `onGetLocalUserInfo` so it only allows the developer to return a picture Url. Maybe simply change the event into `onGetLocalUserPicture` and have it return a string.

Or, we may even simply drop the ability to display a picture: why is it important to enforce a non-spoofable name while at the same time not enforce a non-spoofable picture?

Reply:
discussed in meeting, came up with a solution where IUserInfo extends IEventUserInfo, instead of IUserInfo being a type union.